### PR TITLE
MRG: Revert to using dict subclass for Info

### PR DESCRIPTION
**Pull request by Eric89GXL**
Resets us to using `dict` subclass for `Info` for 0.12 before we tackle #2860 for 0.13 (or later).

Ready for review/merge from my end.

### CODE REVIEW

**Comment by agramfort** [comment]
LGTM

@wmvanvliet can you double check?

**Comment by wmvanvliet** [comment]
LGTM. +1 for merge.

**Comment by Eric89GXL** [comment]
Thanks for the reviews @agramfort @wmvanvliet. @dengemann you have time to take a look?

File: mne/io/array/array.py [hunk 1]
```diff
@@ -38,7 +38,6 @@ def __init__(self, data, info, first_samp=0, verbose=None):
         if not isinstance(info, Info):
             raise TypeError('info must be an instance of Info, got %s'
                             % type(info))
-        info._check_consistency()  # make sure it's valid
```
Comment (agramfort) [inline mne/io/array/array.py]: why?

**Comment by agramfort** [comment]
otherwise +1 for merge

**Comment by agramfort** [comment]
@jaeilepp will review tomorrow :)

File: mne/io/array/array.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/io/array/array.py]: Super class `_BaseRaw` calls it in its constructor shortly thereafter

**Comment by dengemann** [comment]
Thanks, this looks good. So we have the consistency checking and updating that @wmvanvliet elaborated while keeping things flat. Wonderful! Also +1 from my side.

**Comment by agramfort** [comment]
great !

File: mne/io/kit/kit.py [hunk 2]
```diff
@@ -208,6 +208,7 @@ def _set_stimchannels(self, info, stim, stim_code):
             chan_info['loc'] = np.zeros(12)
             chan_info['kind'] = FIFF.FIFFV_STIM_CH
             info['chs'].append(chan_info)
+            info._update_redundant()
```
Comment (jaeilepp) [inline mne/io/kit/kit.py]: Shouldn't this be called even if ``stim is None``

File: mne/viz/raw.py [hunk 3]
```diff
@@ -61,7 +61,8 @@ def _update_raw_data(params):
         data[di] /= params['scalings'][params['types'][di]]
         # stim channels should be hard limited
         if params['types'][di] == 'stim':
-            data[di] /= float(max(data[di]))
+            norm = float(max(data[di]))
+            data[di] /= norm if norm > 0 else 1.
```
Comment (jaeilepp) [inline mne/viz/raw.py]: What are negative stim values used for? Shouldn't they be also scaled?

Comment (Eric89GXL) [inline mne/viz/raw.py]: this is a safety against `norm == 0` mostly

Comment (Eric89GXL) [inline mne/viz/raw.py]: (i.e., no values on stim channel)

File: mne/io/kit/kit.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/io/kit/kit.py]: no, it only needs to be called (again) if `chs` is updated and it's not updated if `stim is None`

Comment (Eric89GXL) [inline mne/io/kit/kit.py]: It already gets called before this earlier by `get_kit_info` on line 738

File: mne/viz/raw.py (hunk 3 above)
Comment (jaeilepp) [inline mne/viz/raw.py]: Ok. I see what you mean.

**Pull request merged**